src/item_factory.cpp

```diff
@@ -414,6 +414,18 @@ void Item_factory::finalize_post( itype &obj )
             }
         }
     }
+
+    if( !obj.ascii_picture.empty() ) {
+        std::vector<std::string> tmp_ascii_pic;
+        for( std::string line : obj.ascii_picture ) {
+            if( line.length() > 42 ) {
```

Comment:
You can't really use neither `length` nor `substr` here. These functions consider string length in terms of bytes, but the string is actually in UTF-8 (multi-byte encoding) and can also contain tags. Use `utf8_width` and `remove_color_tags` to get the actual "printable" length of the string.

`42` is a magic number (quite famous, but still magic). Use a named constant instead.